python36Packages.cvxopt: 1.2.1 -> 1.2.2

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/python3.6-cvxopt/versions.



meta.description for python36Packages.cvxopt is: '"Python Software for Convex Optimization"'.


<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 1.2.2 with grep in /nix/store/qzrxzj1q6lrph2dzqz6g3mzg9l8nvx53-python3.6-cvxopt-1.2.2
- directory tree listing: https://gist.github.com/64fc18d3e537910ac2ff18cb3f42455f
- du listing: https://gist.github.com/d2dcc00af379a7c5f1fd3444bbfa1afd

</details>
<details>
<summary>
Outpath report (click to expand)
</summary>

Outpath difference report
10 total rebuild paths

4 package rebuilds

4 x86_64-linux rebuilds
2 i686-linux rebuilds
2 x86_64-darwin rebuilds
2 aarch64-linux rebuilds


First ten rebuilds by attrpath
python27Packages.cvxopt
python36Packages.cvxopt
sage
sageWithDoc

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/qzrxzj1q6lrph2dzqz6g3mzg9l8nvx53-python3.6-cvxopt-1.2.2 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A python36Packages.cvxopt https://github.com/r-ryantm/nixpkgs/archive/2df88cb6f93d55a72c8f660a4fc063c9f7a56a88.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/qzrxzj1q6lrph2dzqz6g3mzg9l8nvx53-python3.6-cvxopt-1.2.2
ls -la /nix/store/qzrxzj1q6lrph2dzqz6g3mzg9l8nvx53-python3.6-cvxopt-1.2.2/bin
```


</details>
<br/>


cc @edwtjo for testing.

r-ryantm committed:
python36Packages.cvxopt: 1.2.1 -> 1.2.2

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/python3.6-cvxopt/versions

GrahamcOfBorg commented:
<!--REQUEST_ID=9f1f8c6e-1720-4d03-92e0-c0bb91a13b89-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=23cdadd2-3174-4589-8243-759bcc68390f)

Attempted: python36Packages.cvxopt

<details><summary>Partial log (click to expand)</summary><p>

```

22: -3.1548e+02 -3.1553e+02  7e-01  1e-06  8e-05  1e-03
23: -3.1537e+02 -3.1540e+02  5e-01  7e-07  5e-05  9e-04
24: -3.1536e+02 -3.1540e+02  9e-01  9e-07  7e-05  2e-03
25: -3.1533e+02 -3.1536e+02  5e-01  7e-07  6e-05  1e-03
26: -3.1515e+02 -3.1517e+02  3e-01  3e-07  2e-05  7e-04
27: -3.1503e+02 -3.1504e+02  3e-02  3e-08  2e-06  8e-05
28: -3.1502e+02 -3.1502e+02  3e-04  3e-10  3e-08  8e-07
29: -3.1502e+02 -3.1502e+02  3e-06  3e-12  3e-10  8e-09
Optimal solution found.
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=9f1f8c6e-1720-4d03-92e0-c0bb91a13b89-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=c98f24c9-f81e-4030-9fcd-cd2b3f17490f)

Attempted: python36Packages.cvxopt

<details><summary>Partial log (click to expand)</summary><p>

```
removing redundant constraint 'LF1100B4'
removing redundant constraint 'LF1100B5'
removing redundant constraint 'LF1102B1'
removing redundant constraint 'LF1102B2'
removing redundant constraint 'LF1102B3'
removing redundant constraint 'LF1102B4'
removing redundant constraint 'LF1200B1'
removing redundant constraint 'LF1201B1'
builder for '/nix/store/zx2sg6xf9ifkwsp7wsj7vya6i1k30023-python3.6-cvxopt-1.2.2.drv' failed with exit code 1
error: build of '/nix/store/zx2sg6xf9ifkwsp7wsj7vya6i1k30023-python3.6-cvxopt-1.2.2.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=9f1f8c6e-1720-4d03-92e0-c0bb91a13b89-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=e433bdba-9590-4990-af57-d387b05a059c)

Attempted: python36Packages.cvxopt

<details><summary>Partial log (click to expand)</summary><p>

```
21: -3.1605e+02 -3.1621e+02  2e+00  4e-06  3e-04  5e-03
22: -3.1548e+02 -3.1553e+02  7e-01  1e-06  8e-05  1e-03
23: -3.1537e+02 -3.1540e+02  5e-01  7e-07  5e-05  9e-04
24: -3.1536e+02 -3.1540e+02  9e-01  9e-07  7e-05  2e-03
25: -3.1533e+02 -3.1536e+02  5e-01  7e-07  6e-05  1e-03
26: -3.1515e+02 -3.1517e+02  3e-01  3e-07  2e-05  7e-04
27: -3.1503e+02 -3.1504e+02  3e-02  3e-08  2e-06  8e-05
28: -3.1502e+02 -3.1502e+02  3e-04  3e-10  3e-08  8e-07
29: -3.1502e+02 -3.1502e+02  3e-06  3e-12  3e-10  8e-09
Optimal solution found.
```
</p></details>

timokau commented:
Darwin failure is not a regression: https://hydra.nixos.org/job/nixpkgs/trunk/python36Packages.cvxopt.x86_64-darwin

CC @jbaum98 who I think originally added darwin support.

timokau commented:
@NixOS/darwin-maintainers anyone interested in debugging the cvxopt darwin failure or should I just disable support?

LnL7 commented:
Looks like the libraries in `suiteparse` are the actual issue, the `install_name` are pointing to the build directory instead of the output.

LnL7 committed:
suitesparse: fixup darwin libraries

The build created libraries with an install_name that points to the
build directory instead of the installation prefix.  Causing errors like
this when other packages try to link against it's libraries.

    Library not loaded: /private/tmp/nix-build-suitesparse-5.3.0.drv-0/SuiteSparse/lib/libcholmod.3.0.12.dylib

GrahamcOfBorg commented:
<!--REQUEST_ID=7f9a4ace-49af-4f4a-bfb3-e5c11213c16b-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=9f36deee-bf59-483f-a801-2b9ef1b5c070)

Attempted: python36Packages.cvxopt, suitesparse

<details><summary>Partial log (click to expand)</summary><p>

```
21: -3.1605e+02 -3.1621e+02  2e+00  4e-06  3e-04  5e-03
22: -3.1548e+02 -3.1553e+02  7e-01  1e-06  8e-05  1e-03
23: -3.1537e+02 -3.1540e+02  5e-01  7e-07  5e-05  9e-04
24: -3.1536e+02 -3.1540e+02  9e-01  9e-07  7e-05  2e-03
25: -3.1533e+02 -3.1536e+02  5e-01  7e-07  6e-05  1e-03
26: -3.1515e+02 -3.1517e+02  3e-01  3e-07  2e-05  7e-04
27: -3.1503e+02 -3.1504e+02  3e-02  3e-08  2e-06  8e-05
28: -3.1502e+02 -3.1502e+02  3e-04  3e-10  3e-08  8e-07
29: -3.1502e+02 -3.1502e+02  3e-06  3e-12  3e-10  8e-09
Optimal solution found.
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=7f9a4ace-49af-4f4a-bfb3-e5c11213c16b-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=2f46452b-3560-43e4-b0bd-c43c7e708177)

Attempted: python36Packages.cvxopt, suitesparse

<details><summary>Partial log (click to expand)</summary><p>

```
23: -3.1537e+02 -3.1540e+02  5e-01  7e-07  5e-05  9e-04
24: -3.1536e+02 -3.1540e+02  9e-01  9e-07  7e-05  2e-03
25: -3.1533e+02 -3.1536e+02  5e-01  7e-07  6e-05  1e-03
26: -3.1515e+02 -3.1517e+02  3e-01  3e-07  2e-05  7e-04
27: -3.1503e+02 -3.1504e+02  3e-02  3e-08  2e-06  8e-05
28: -3.1502e+02 -3.1502e+02  3e-04  3e-10  3e-08  8e-07
29: -3.1502e+02 -3.1502e+02  3e-06  3e-12  3e-10  8e-09
Optimal solution found.
/nix/store/11zvbpdga7yh2mkx6jb5acax1xylim8i-python3.6-cvxopt-1.2.2
/nix/store/ggh3f0kgm987ccryhwxspzzal0jr6jrb-suitesparse-5.3.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=7f9a4ace-49af-4f4a-bfb3-e5c11213c16b-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=e82e4a97-631e-4146-80c5-d19030cb2298)

Attempted: python36Packages.cvxopt, suitesparse

<details><summary>Partial log (click to expand)</summary><p>

```
removing redundant constraint 'LF1100B4'
removing redundant constraint 'LF1100B5'
removing redundant constraint 'LF1102B1'
removing redundant constraint 'LF1102B2'
removing redundant constraint 'LF1102B3'
removing redundant constraint 'LF1102B4'
removing redundant constraint 'LF1200B1'
removing redundant constraint 'LF1201B1'
builder for '/nix/store/db1wan6jvq090l742q0lx8jymwqys4l1-python3.6-cvxopt-1.2.2.drv' failed with exit code 1
error: build of '/nix/store/db1wan6jvq090l742q0lx8jymwqys4l1-python3.6-cvxopt-1.2.2.drv' failed
```
</p></details>

LnL7 commented:
Hrm, looks like that's too late since the libraries reference each other. Not sure how much work it is to fix the build system, previous versions worked fine tho.

timokau committed:
pythonPackages.cvxopt: mark as broken on darwin

See https://github.com/NixOS/nixpkgs/pull/49994#issuecomment-437589336.

GrahamcOfBorg commented:
<!--REQUEST_ID=917a0c79-8951-4a55-bd43-ceb0a23e005f-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=e42aaded-ac56-4c9f-8d8c-2743750d9638)

Attempted: suitesparse

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (14.28 MiB download, 68.82 MiB unpacked):
  /nix/store/1mr5kl3nx74plgwyp99lmjr08x13m1zw-gfortran-7.3.0-lib
  /nix/store/j0fbs9gz178cknpzrpvngald2ca5msvy-suitesparse-5.3.0
  /nix/store/k7q2kh3mhfsd53wiy0fyxcm02klq2i0k-openblas-0.3.3
  /nix/store/wcg2bimzqhqza2vz86hbf2q904pkfy85-glibc-2.27
copying path '/nix/store/wcg2bimzqhqza2vz86hbf2q904pkfy85-glibc-2.27' from 'https://cache.nixos.org'...
copying path '/nix/store/1mr5kl3nx74plgwyp99lmjr08x13m1zw-gfortran-7.3.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/k7q2kh3mhfsd53wiy0fyxcm02klq2i0k-openblas-0.3.3' from 'https://cache.nixos.org'...
copying path '/nix/store/j0fbs9gz178cknpzrpvngald2ca5msvy-suitesparse-5.3.0' from 'https://cache.nixos.org'...
/nix/store/j0fbs9gz178cknpzrpvngald2ca5msvy-suitesparse-5.3.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=917a0c79-8951-4a55-bd43-ceb0a23e005f-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=088e5626-d9b7-4771-a4ab-bd2d1507d442)

Attempted: suitesparse

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/libldl.2.2.6.dylib: fixing dylib
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/libmetis.dylib: fixing dylib
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/librbio.2.2.6.dylib: fixing dylib
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/libspqr.2.0.8.dylib: fixing dylib
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/libsuitesparse.dylib: fixing dylib
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/libsuitesparseconfig.5.3.0.dylib: fixing dylib
/nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib/libumfpack.5.7.7.dylib: fixing dylib
strip is /nix/store/rk3p99jbj4idl3dkkg1rxja5gz0qgs98-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0/lib
patching script interpreter paths in /nix/store/rpy2fxqjx1p9znl02qs8zjr982a6icl0-suitesparse-5.3.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=917a0c79-8951-4a55-bd43-ceb0a23e005f-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49994&attempt_id=206af935-214c-4632-8f98-e3991aaa78aa)

Attempted: suitesparse

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib/libcamd.so.2.4.6
shrinking /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib/libccolamd.so.2.9.6
shrinking /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib/libbtf.so.1.2.6
shrinking /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib/libamd.so.2.4.6
shrinking /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib/libmetis.so
shrinking /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib/libsuitesparseconfig.so.5.3.0
strip is /nix/store/6dpnd5aniypn8124mmy8f88s4mq2zl07-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0/lib
patching script interpreter paths in /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0
checking for references to /build in /nix/store/i5iqdd4y450x69idx7xh9f75gasgndjn-suitesparse-5.3.0...
```
</p></details>
